src/widget/button.rs

```diff
@@ -77,6 +83,8 @@ pub struct Image {
     pub color: ImageColor,
     /// The rectangular area of the original source image that should be displayed.
     pub src_rect: Option<Rect>,
+    /// The id of an Alternate image to show when pressed
+    pub alt_id: Option<widget::Id>,
```

Comment:
Perhaps a better name field for this would be `press_id`, to clarify that it is the image used while the button is pressed.